### ISSUE TITLE
Cannot define function-like macros via configure.py

### ISSUE THREAD

**Issue by akosthekiss** [svaarala/duktape#2013]
Configuration of `DUK_USE_FATAL_HANDLER` and `DUK_USE_DEBUG_WRITE` does not work as suggested in [wiki](https://wiki.duktape.org/Configuring2x.html):

```
$ python2 tools/configure.py --output-directory=build '-DDUK_USE_FATAL_HANDLER(udata,msg)=do { const char *fatal_msg = (msg); fprintf(stderr, "*** FATAL ERROR: %s\n", fatal_msg ? fatal_msg : "no message"); abort(); } while (0)' -DDUK_USE_ASSERTIONS -DDUK_USE_SELF_TESTS -DDUK_USE_DEBUG -DDUK_USE_DEBUG_LEVEL=0 '-DDUK_USE_DEBUG_WRITE(level,file,line,func,msg)=do {fprintf(stderr, "D%ld %s:%ld (%s): %s\n", (long) (level), (file), (long) (line), (func), (msg));} while(0)'
Traceback (most recent call last):
  File "tools/configure.py", line 997, in <module>
    main()
  File "tools/configure.py", line 257, in main
    (opts, args) = parser.parse_args()
  File "/usr/lib/python2.7/optparse.py", line 1400, in parse_args
    stop = self._process_args(largs, rargs, values)
  File "/usr/lib/python2.7/optparse.py", line 1444, in _process_args
    self._process_short_opts(rargs, values)
  File "/usr/lib/python2.7/optparse.py", line 1551, in _process_short_opts
    option.process(opt, value, values, self)
  File "/usr/lib/python2.7/optparse.py", line 789, in process
    self.action, self.dest, opt, value, values, parser)
  File "/usr/lib/python2.7/optparse.py", line 809, in take_action
    self.callback(self, opt, value, parser, *args, **kwargs)
  File "duktape/tools/genconfig.py", line 1391, in add_force_option_define
    raise Exception('invalid option value: %r' % value)
Exception: invalid option value: 'DUK_USE_FATAL_HANDLER(udata,msg)=do { const char *fatal_msg = (msg); fprintf(stderr, "*** FATAL ERROR: %s\\n", fatal_msg ? fatal_msg : "no message"); abort(); } while (0)'
```

Actually, there are two problems with the current processing approach:
- Current code doesn't allow an equal sign to appear in the value part of the definition (that's the cause for the error message above).
- If that's worked around, current code still cannot deal with the arguments of a function-like macro.

**Comment by svaarala** [comment]
That's odd, I'm fairly sure it has worked at some point previously. I'll look into it, but meanwhile, you can place the definitions in a config file instead to work around the issue.

**Comment by akosthekiss** [comment]
FYI: I've proposed a fix in #2014

**Comment by svaarala** [comment]
The fix looks good and is merged now, thanks! I don't think it could have worked before so I must just be remembering wrong. Oh well :)
